volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -581,3 +587,262 @@ def reference_count(self):
             ret = refcnt.refs.counter
 
         return ret
+
+class mnt_namespace(objects.StructType):
+    def get_inode(self):
+        if self.has_member("proc_inum"):
+            return self.proc_inum
+        elif self.ns.has_member("inum"):
+            return self.ns.inum
+        else:
+            raise AttributeError("Unable to find mnt_namespace inode")
+
+class net(objects.StructType):
+    def get_inode(self):
+        if self.has_member("proc_inum"):
+            return self.proc_inum
+        elif self.ns.has_member("inum"):
+            return self.ns.inum
+        else:
+            raise AttributeError("Unable to find net_namespace inode")
+
+class socket(objects.StructType):
+    def _get_vol_kernel(self):
+        symbol_table_arr = self.vol.type_name.split("!", 1)
+        symbol_table = symbol_table_arr[0] if len(symbol_table_arr) == 2 else None
+
+        module_names = list(self._context.modules.get_modules_by_symbol_tables(symbol_table))
+        if not module_names:
+            raise ValueError(f"No module using the symbol table {symbol_table}")
+
+        kernel_module_name = module_names[0]
+        kernel = self._context.modules[kernel_module_name]
+        return kernel
+
+    def get_inode(self):
+        try:
+            kernel = self._get_vol_kernel()
+        except ValueError:
+            return 0
```

Comment:
This seems like you're making a decision that the caller might want to make?  Can an inode ever be 0, or is it always an error when it happens?